keps/sig-scheduling/20200612-simplified-topology-manager.md

```diff
@@ -0,0 +1,229 @@
+---
+title: Simplified version of TopologyManager in kube-scheduler
+authors:
+  - "@AlexeyPerevalov"
+  - "@swatisehgal"
+owning-sig: sig-scheduling
+participating-sigs:
+reviewers:
+  - "@ahg-g"
+  - "@huang-wei"
+  - "@derekwaynecarr"
+approvers:
+  - "@ahg-g"
+  - "@huang-wei"
+editor: TBD
+creation-date: 2020-05-28
+last-updated: 2020-05-28
+status: implementable
+see-also:
+superseded-by:
+---
+# Topology Aware Scheduling
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Topology format](#topology-format)
+  - [Plugin implementation details](#plugin-implementation-details)
+    - [Topology information in the TopologyMatch plugin](#topology-information-in-the-topologymatch-plugin)
+    - [Description of the Algorithm](#description-of-the-algorithm)
+  - [Accessing NodeResourceTopology CRD](#accessing-noderesourcetopology-crd)
+- [Use cases](#use-cases)
+- [Test plans](#test-plans)
+- [Graduation criteria](#graduation-criteria)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature enablement and rollback](#feature-enablement-and-rollback)
+- [Implementation history](#implementation-history)
+<!-- /toc -->
+
+# Summary
+
+This document describes behavior of the Kubernetes Scheduler which take into
+account worker node NUMA topology.
+
+# Motivation
+
+After Topology Manager was introduced the problem of launching pod in the
+cluster where worker nodes have different NUMA topology and different amount
+of resources in that topology became actual. Pod could be scheduled on the node
+where total amount of resources are enough, but resource distribution could not
+satisfy the appropriate Topology policy. In this case the pod failed to start. Much
+better behaviour for scheduler would be to select appropriate node where kubelet admit
+handlers may pass.
+
+
+## Goals
+
+- Make scheduling process more precise when we have NUMA topology on the
+worker node.
+
+## Non-Goals
+
+- Change the PodSpec to allow requesting a specific node topology manager policy
+- This Proposal requires exposing NUMA topology information. This KEP doesn't
+describe how to expose all necessary information it just declare what kind of
+information is necessary.
+
+# Proposal
+Kube-scheduler builtin plugin will be added to the main tree. This plugin
```

Comment:
are you advocating for kubelet itself to use this plugin?

Reply:
Kubelet does not need to use the scheduler plugin. The scheduler plugin runs a simplified version of topology manager logic to filter out nodes that can't align resources on a single numa node (in case single -numa-node policy). Once a suitable node is identified, the kubelet behaves as normal allocating resources, topology managers aligns resources based on numa nodes.